### PR TITLE
(#257) - Use FileInDir wherever possible

### PR DESCRIPTION
**Pull request by victornoel**
This is for #257. I've added extra tests also to validate my understanding of the expected behaviour of the class.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#258](https://github.com/artipie/rpm-adapter/issues/258) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/a0494e98-de90-4cc1-909b-8832e7ac68d9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/rpm/misc/FileInDirTest.java [hunk 1]
```diff
@@ -46,4 +47,24 @@ void findsFile(@TempDir final Path tmp) throws IOException {
             new IsEqual<>(file)
         );
     }
+
+    @Test
+    void doesNotFindFile(@TempDir final Path tmp) throws IOException {
+        final Path file = tmp.resolve("a_file.txt");
+        Files.write(file, "abc123".getBytes());
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/misc/FileInDirTest.java:54]: @victornoel You don't need to declare `file`, you could inline it into `Files.write` call

File: src/test/java/com/artipie/rpm/misc/FileInDirTest.java [hunk 2]
```diff
@@ -46,4 +47,24 @@ void findsFile(@TempDir final Path tmp) throws IOException {
...
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () -> new FileInDir(tmp).find("_notfile.t")
+        );
+    }
+
+    @Test
+    void doesNotUseRegexToFind(@TempDir final Path tmp) throws IOException {
+        final Path file = tmp.resolve("fileXtxt");
+        Files.write(file, "ab123".getBytes());
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/misc/FileInDirTest.java:64]: @victornoel Here too, please inline `file`

**Comment by victornoel** [comment]
@paulodamaso it is done, thanks. I have verified and didn't find any place where this would be needed.

File: src/test/java/com/artipie/rpm/misc/FileInDirTest.java [hunk 3]
```diff
@@ -46,4 +47,22 @@ void findsFile(@TempDir final Path tmp) throws IOException {
...
+    void doesNotFindFile(@TempDir final Path tmp) throws IOException {
+        Files.write(tmp.resolve("a_file.txt"), "abc123".getBytes());
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () -> new FileInDir(tmp).find("_notfile.t")
+        );
+    }
+
+    @Test
+    void doesNotUseRegexToFind(@TempDir final Path tmp) throws IOException {
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/misc/FileInDirTest.java:61]: @victornoel Why `FileInDir` shouldn't use regex to find a file?

Comment (victornoel) [reply src/test/java/com/artipie/rpm/misc/FileInDirTest.java:61]: @paulodamaso I don't know, that's how it is currently implemented at least. If this should change, we can add a todo if you want? Not sure it is desired though…

**Comment by paulodamaso** [comment]
@g4s8 please merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**